### ISSUE TITLE
Gin authentication

### ISSUE THREAD

**Issue by apachejuice** [deepmap/oapi-codegen#834]
Is it possible to use `securitySchemes` with gin? It appears there are only functions to create echo middleware for this. What I found from the documentation is something along the lines of this:
```go
validator := middleware.OapiRequestValidatorWithOptions(
	spec,
	&middleware.Options{
		Options: openapi3filter.Options{
			AuthenticationFunc: authFn,
		},
	},
)
```
The obvious issue here is that `validator` is of type `echo.MiddlewareFunc` which ostensibly does not work with gin routers. How to do this with gin?

**Comment by jamietanna** [comment]
Yep, there's Gin middleware which should work with what you want? ([example](https://github.com/jamietanna/oapi-codegen/blob/5a103c50bc54c0ba99e2a0a1853dbd411628eeb5/examples/petstore-expanded/gin/petstore.go#L35))

**Comment by mromaszewicz** [comment]
Gin middleware for authentication is available. See the examples/petstore-expanded/gin/ example. Closing as answered.
